core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -922,7 +922,7 @@ private[spark] class MapOutputTrackerMaster(
         Runtime.getRuntime.availableProcessors(),
         statuses.length.toLong * totalSizes.length / parallelAggThreshold + 1).toInt
       if (parallelism <= 1) {
-        for (s <- statuses) {
+        statuses.filter(_ != null).foreach { s =>
```

Comment:
This looks like an unnecessary change. Could `status` be null after this PR?

Reply:
Based on this [comment](https://github.com/apache/spark/pull/33896#discussion_r744193682), we would need this `null` filtering right?